README.md
This is a little baseball simulation that simulates a game between two made up teams.
Simulate a new, unique game every time you reload the page! 

Features:
- Two teams compete in nine innings
- Players each have individualized stats that determine their performance/probabilities
- Each pitch is simulated in-depth with a series of probabilities
- Clean, organized  user interface with all kinds of stats/information

Finally, the project is finished! Check out 1.0.0 at: https://mydigitalspot.dev/simulation or download the file from the repo page. 

For a changelog, refer to the CHANGELOG.txt file or see the file history on Github. 

Thanks for checking out this project, and I hope you have some fun using it! @chillzeus

CHANGELOG.txt

<- CHANGELOG: ->
(this is all kind of arbitrary up until the full release, there's no logic to this versioning)
(basically it just lets me keep track of the progress of this project and stuff)

FULL RELEASE:
1.0.0 - Oct 4, 2022

PARTIAL RELEASE/EARLY ACCESS: 

0.1.4:
- Extra innings finally implemented
- Introduces a new "In-depth statistics" section
- Fixed minor bugs

0.1.3:
- Simplified hitting probabilities (stuff was getting out of control)
- Added more UI elements
    - MVP display
    - Pitching stats
- Changed up the player ratings
- Fixed error where game was 4.5 innings

0.1.2: 
- More UI stuff like:
- Fixed error where team would play against themselves
    - Player rosters with their individual stats
    - Stats for the pitchers
- Implemented more in-depth hit/strikeout/walk stats for each individual player
- Added a relief pitcher for each team

0.1.1:
- Started user interface:
    - Scoreboard
    - Basic stats

0.1.0:
- Simulation works/has most features but issues such as:
    - Way too few pitches per inning
    - Doesn't cycle through pitchers
- No user interface yet, coming soon-ish
- Added README stuff (like changelog)


EARLY DEVELOPMENT:

0.0.11:
- Fixed glitch where team 2 always won
- Added more functionality to the console

0.0.10:
- The two teams now play each other
- Added more players

0.0.9:
- Brought down runs scored to normal levels
- Added innings
- Organized console

0.0.8:
- Added outs, fills bases, etc
- Probability factors

0.0.7:
- Probabilities
- Improved stuff to test probabilities

0.0.6:
- Stuff to test out probabilities

0.0.5:
- Tweaked a few things

0.0.4:
- Does not work as intended
- Attempted to add a different probability system

0.0.3:
- Basic probabilities

0.0.2:
- Adds bases, balls and strikes

0.0.1 (includes all previous commits):
- Created necessary files
- Hit system
- Started all the probabilties

simulator.js
const maxEvents = 100;
const defaultProbability = 22;
const defaultHitType = 35;

const defaultTestRuns = 625;
const defaultTestFactor = 10;

function atBat(batter, pitcher) {
    // This keeps us from doing too much in one at-bat.
    // You can remove it after testing since foul balls
    //  will cause at-bats to go longer, but they'll
    //  never hit a threshold like 100.
    let events = 0;

    let balls = 0;
    let strikes = 0;

    // This is not actually needed. We could say while(true) { ... } and the
    //  return statements will handle exiting the loop.
    let atBatDone = false;

    // Handle all the probability adjustments in a separate function.
    const probability = getProbabilityOffset(batter, pitcher, defaultProbability);

    while (!atBatDone) {
        events++;
        // Get the result from a pitch, then handle that result in the next section of logic.
        const pitchResult = getPitchResult(probability);

        if (pitchResult === 'B') balls++;
        else if (pitchResult === 'K') strikes++;
        else if (pitchResult === 'IP') {
            atBatDone = true;
            return getInPlayResult(batter, pitcher);
        }

        if (balls >= 4) {
            console.log("Walk");
            atBatDone = true;
            return "BB";
        }
        if (strikes >= 3) {
            console.log("Strikeout");
            atBatDone = true;
            return "K";
        }
        // This shouldn't happen, but you never know.
        if (events >= maxEvents) {
            console.log("Error");
            atBatDone = true;
            return "ERR";
        }
    }
}

function getProbabilityOffset(batter, pitcher, startingProbability) {
    // It's best practice to not modify parameters in a function.
    // This is an example of a "side-effect" and isn't great since
    //  things can change in other places and it gets confusing.
    let probability = startingProbability;
    //sees who is the "better player"
    if (pitcher.pitchingRating > batter.battingRating) {
        probability += 5;
    }
    if (pitcher.pitchingRating < batter.battingRating) {
        probability += -2;
    }
    if (pitcher.handed !== batter.handed) {
        probability += -1;
    }
    if (pitcher.exhaustion === 2) {
        probability += -1;
    }
    if (pitcher.exhaustion === 3) {
        probability += -2;
    }

    return probability;
}

function getPitchResult(probability) {
    const randomNumber = Math.trunc(Math.random() * probability);
    if (randomNumber <= 6) {
        console.log("in play")
        return "IP";
    }
    if (randomNumber >= 7 && randomNumber <= 17) {
        console.log("ball")
        return "B";
    }
    if (randomNumber >= 18) {
        console.log("strike")
        return "K";
    }
}

function getInPlayResult(batter, pitcher) {
    let hitType = defaultHitType;

    if (batter.strength > 1 && batter.strength < 1.5) {
        hitType += 3;
    }
    if (batter.strength > 1.5) {
        hitType += 4;
    }

    const randomHit = Math.trunc(Math.random() * hitType);

    if (randomHit <= 12) {
        console.log("ground out")
        return "GO"
    }
    if (randomHit <= 22) {
        console.log("fly out")
        return "FO";
    }
    if (randomHit <= 28) {
        console.log("single");
        return "1B";
    }
    if (randomHit <= 31) {
        console.log("double");
        return "2B";
    }
    if (randomHit === 32) {
        console.log("triple");
        return "3B";
    }
    if (randomHit >= 33) {
        console.log("home run");
        return "HR";
    }
}

function runMultipleAtBats(inputTestRuns, inputTestFactor, batter, pitcher) {
    // Initialize our object of results
    const results = {};

    // We can either use the values entered in the function
    //  or fall back to the defaults set elsewhere.
    const testRuns = inputTestRuns || defaultTestRuns;
    const testFactor = inputTestFactor || defaultTestFactor;

    // Run the specified number of at-bats.
    // After each at-bat, we increase the result total by 1
    for (let run = 0; run < (testRuns * testFactor); run++) {
        const result = atBat(batter, pitcher);

        results[result] = (results[result] || 0) + 1
        console.log(`Result = [${result}]`);
    }

    // We go through each key/value entry in the `results` object and divide
    //  the total by the number of times we went through the test runs.
    // Think of this like playing a number of seasons with testFactor as that number.
    const factoredResults = Object.entries(results).reduce((r, [key, total]) => {
        r[key] = parseInt(total / testFactor);
        return r;
    }, results);

    console.table(factoredResults);

    const stats = getStats(factoredResults, testRuns);

    console.log(`Completed ${testFactor} sets of ${testRuns} at bats.`);

    console.table(stats);

    return stats;
}

function getStats(results, testRuns) {
    // Get all the stats or zero if that stat isn't present in the results.
    const singles = (results['1B'] || 0);
    const doubles = (results['2B'] || 0);
    const triples = (results['3B'] || 0);
    const homeRuns = (results['HR'] || 0);
    const walks = (results['BB'] || 0);

    const hits = (singles + doubles + triples + homeRuns);
    const atBats = (testRuns - walks);

    // AVG = Batting average (Percent of time getting a hit)
    // OBP = On-base percentage (Percent of time getting on base)
    // SLG = Slugging percentage (Average bases per at bat)
    // OPS = On-base percentage + slugging percentage (Math!)
    const avg = hits / atBats;
    const obp = ((hits + walks) / testRuns);
    const slg = ((singles + doubles + doubles + triples + triples + triples + (homeRuns * 4))/atBats);
    const ops = obp + slg;

    return {
        pa: testRuns,
        ab: atBats,
        hits: hits,
        avg: avg.toFixed(3),
        obp: obp.toFixed(3),
        slg: slg.toFixed(3),
        ops: ops.toFixed(3)
    };
}

// This is a node.js convention that allows you to use functions/objects in other places in your app.
module.exports = {
    atBat,
    runMultipleAtBats
}

script.js
// 1.0.0.2

let balls = 0;
let strikes = 0;

let firstbase = false;
let secondbase = false;
let thirdbase = false;

let probability = 25;

let randomNumber = null;
let randomHit = null;
let secondary = null;
let inPlay = false;

let out = 0;
let nextplay = false;
let atBats = 0;
let inning = 0;
let hitType = 100;

//player[i].etc USE FOR LOOP!
let i = 0;
let z = 0;

const team1 = [
  {
  "runs": 0,
  "single": 0,
  "double": 0,
  "triple": 0,
  "homeRun": 0,
  "strikeout": 0,
  "walk": 0,
  "atBats": 0,
  "pitches": 0,
  "strikes": 0,
  "hits": 0,
    "players": [
    {
        "name": "Soren Fiedler",
        "rating": 5,
        "strength": 1.4,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Isaac Fiedler",
        "rating": 7,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Adam Fiedler",
        "rating": 9,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Kristina Fiedler",
        "rating": 4,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Andrew Rothe",
        "rating": 6,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Sara Doering-Fiedler",
        "rating": 4,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Greta Fiedler",
        "rating": 5,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Jayde Doering",
        "rating": 6,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Mary Doering",
        "rating": 4,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    }

    ],
    "pitchers": [ 
    {
      "name": "Rosemary Fiedler",
      "rating": 4,
      "pitches": 0,
      "strikes": 0,
      "strikeout": 0,
      "walk": 0,
    },
    {
      "name": "Zeus the Frog",
      "rating": 8,
      "pitches": 0,
      "strikes": 0,
      "strikeout": 0,
      "walk": 0,
    }
    ],
  }
]

const team2 = [
  {
  "runs": 0,
  "single": 0,
  "double": 0,
  "triple": 0,
  "homeRun": 0,
  "strikeout": 0,
  "walk": 0,
  "atBats": 0,
  "pitches": 0,
  "strikes": 0,
  "hits": 0,
    "players": [
    {
        "name": "Ron Doering",
        "rating": 5,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Harriet Rusk",
        "rating": 3,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Kevin Rusk",
        "rating": 6,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Mike Doering",
        "rating": 8,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Monika Fiedler",
        "rating": 7,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Alexander Doering",
        "rating": 4,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Ellie Doering",
        "rating": 5,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },
    {
        "name": "Chipotle the Fish",
        "rating": 3,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    },   
    {
        "name": "Kathy Rothe",
        "rating": 7,
        "hits": 0,
        "single": 0,
        "double": 0,
        "triple": 0,
        "homerun": 0,
        "strikeout": 0,
        "walk": 0,
        "RBI": 0
    }
    ],
    "pitchers": [
    {
      "name": "Tom Fiedler",
      "rating": 5,
      "pitches": 0,
      "strikeout": 0,
      "walk": 0,    
    },
    {
      "name": "Stephen Rothe",
      "rating": 7,
      "pitches": 0,
      "strikes": 0,
      "strikeout": 0,
      "walk": 0,
    }
    ]
  }
]

let current_team = team1;
let pitch_team = team2;

function atBat() {
  hitType = 100;
  current_team[0].atBats += 1;
  atBats += 1;
  if (nextplay) {
    if (i < 9) {
      nextplay = false;
      i += 1;
    }
    if (i == 9) {
      nextplay = false;
      i = 0;
    }
  }
  inPlay = false;

  // ball gets in play less, but when it gets in play, has higher chance of being a hit??
  // if strikes and stuff  
  while (inPlay == false) {
    randomNumber = Math.trunc(Math.random() * (probability - 0) + 0);
    if (randomNumber <= 10) {
      inPlay = true;
      pitch_team[0].pitches += 1;
      pitch_team[0].pitchers[z].pitches += 1;
    } 
    if (randomNumber >= 11 && randomNumber <= 16) {
      balls += 1;
      pitch_team[0].pitches += 1;
      pitch_team[0].pitchers[z].pitches += 1;
    }
    if (randomNumber >= 17) {
      strikes += 1;
      pitch_team[0].strikes += 1;
      pitch_team[0].pitchers[z].strikes += 1;
      pitch_team[0].pitches += 1;
      pitch_team[0].pitchers[z].pitches += 1;
    }

    //check if walk or strikeout 
    if (balls == 4) {
      balls = 0;
      strikes = 0;
      current_team[0].walk += 1;
      current_team[0].players[i].walk += 1;
      pitch_team[0].pitchers[z].walk += 1;
      nextplay = true;
      atBat();
    } if (strikes == 3) {
      balls = 0;
      strikes = 0;
      current_team[0].strikeout += 1;
      current_team[0].players[i].strikeout += 1;
      pitch_team[0].pitchers[z].strikeout += 1;
      nextplay = true;
      out += 1;
    }
  }

  //the ball is hit into play
  if (inPlay) {

    randomHit = Math.floor(Math.random() * 100) + 1;

    if (randomHit <= 12) {
      secondary = Math.trunc(Math.random() * 100 - 0) + 0;
      if (secondary <= 65) {
        current_team[0].players[i].hits += 1;
        current_team[0].players[i].single += 1;
        current_team[0].single += 1;
        nextplay = true;
        //now we move up the runners
          if (firstbase) {
            secondbase = true;
            firstbase = false;
          } if (secondbase) {
            secondbase = false;
            thirdbase = true;
          } if (thirdbase) {
            thirdbase = false;
            current_team[0].players[i].RBI += 1;
            current_team[0].runs += 1;
          }
        //defines this afterwards to avoid conflicts
        firstbase = true;
        nextplay = true;

        // if it's a double
        } if (secondary > 65 && secondary <= 85) {
          current_team[0].players[i].hits += 1;
          current_team[0].players[i].double += 1;
          current_team[0].double += 1;
          nextplay = true;
          if (firstbase) {
            thirdbase = true;
            firstbase = false;
          } if (secondbase) {
            current_team[0].players[i].RBI += 1;
            current_team[0].runs += 1;
            secondbase = false;
          } if (thirdbase) {
            current_team[0].runs += 1;
            current_team[0].players[i].RBI += 1;
            thirdbase = false;
          }
        secondbase = true;
        nextplay = true;

        // if it's a homerun
        } if (secondary > 85 && secondary <= 98) {
          current_team[0].players[i].hits += 1;
          current_team[0].players[i].homerun += 1;
          current_team[0].homeRun += 1;
          nextplay = true;
          // i know there is probably a better way to write this code
          let runsAdded = 1;
          if (firstbase) {
            runsAdded += 1;
          }
          if (secondbase) {
            runsAdded += 1;
          }
          if (thirdbase) {
            runsAdded += 1;
          }
          current_team[0].runs += runsAdded;
          current_team[0].players[i].RBI += runsAdded;

        // in the event it's a triple
        } if (secondary > 98 && secondary <= 100) {
          current_team[0].triple += 1;
          current_team[0].players[i].hits += 1;
          current_team[0].players[i].triple += 1;
          thirdbase = true;
          nextplay = true;
          if (firstbase) {
            current_team[0].players[i].RBI += 1;
            current_team[0].runs += 1;
            firstbase = false;
          } if (secondbase) {
            current_team[0].players[i].RBI += 1;
            current_team[0].runs += 1;
            secondbase = false;
          } if (thirdbase) {
            current_team[0].runs += 1;
            current_team[0].players[i].RBI += 1;
            thirdbase = false;
          }
        }
      }
    }
    if (randomHit > 65) {
        out += 1;
        nextplay = true;
    }
} 

let determine = 1;
let maxInnings = 18;

function extraInnings() {
  maxInnings += 2;
  console.log("Extra Innings")
}

for (let i = 0; i < 1000; i++) {
  // console.log(out)
  // did we solve the issue with 4 outs?
  if (out >= 3) {
    firstbase = false;
    secondbase = false;
    thirdbase = false;
    out = 0;
    inning += 1;
    if (determine == 1) {
      determine = 2;
      current_team = team2;
      pitch_team = team1;
    }
    else {
      determine = 1;
      current_team = team1;
      pitch_team = team2;
    }
  }
  // switches to reliever
  if (inning == 14) {
    z = 1;
  }
  if (out < 3 && inning <= maxInnings) {
    atBat()
  }
  if (inning == maxInnings) {
    if (team1[0].runs == team2[0].runs) {
      extraInnings();
    }
  }
}

function addUpHits() {
  team1[0].hits = team1[0].single + team1[0].double + team1[0].triple + team1[0].homeRun;
  team2[0].hits = team2[0].single + team2[0].double + team2[0].triple + team2[0].homeRun;
}

addUpHits()

console.log("Simulation, v1.0.0.2")
console.log("Welcome to the simulation console!")

console.log("=========================")

console.log("Team 1 Runs: " + team1[0].runs)
console.log("1B: " + team1[0].single + " 2B: " + team1[0].double + " 3B: " + team1[0].triple + " HR: " + team1[0].homeRun)
console.log("Stikeouts: " + team1[0].strikeout + " Walks: " + team1[0].walk)

console.log("=========================")

console.log("Team 2 Runs: " + team2[0].runs)
console.log("1B: " + team2[0].single + " 2B: " + team2[0].double + " 3B: " + team2[0].triple + " HR: " + team2[0].homeRun)
console.log("Stikeouts: " + team2[0].strikeout + " Walks: " + team2[0].walk)

console.log("=========================")

console.log("Team 1 pitches: " + team1[0].pitches)
console.log("Team 2 pitches: " + team2[0].pitches)


// this section will have all the stuff to be displayed in UI

document.getElementById("team1score").innerHTML = team1[0].runs;
document.getElementById("team2score").innerHTML = team2[0].runs;

document.getElementById("team1-1b").innerHTML = team1[0].single;
document.getElementById("team1-2b").innerHTML = team1[0].double;
document.getElementById("team1-3b").innerHTML = team1[0].triple;
document.getElementById("team1-hr").innerHTML = team1[0].homeRun;
document.getElementById("team1-k").innerHTML = team1[0].strikeout;
document.getElementById("team1-bb").innerHTML = team1[0].walk;

document.getElementById("team2-1b").innerHTML = team2[0].single;
document.getElementById("team2-2b").innerHTML = team2[0].double;
document.getElementById("team2-3b").innerHTML = team2[0].triple;
document.getElementById("team2-hr").innerHTML = team2[0].homeRun;
document.getElementById("team2-k").innerHTML = team2[0].strikeout;
document.getElementById("team2-bb").innerHTML = team2[0].walk;


// gotta say this is rather slick
let m = -1;
function feedInPlayersTeam1() {
  m += 1;
  let feedInPlayers = "<tr><th class='player-name'>" + team1[0].players[m].name + "</th><th>" + team1[0].players[m].hits + "</th><th>" + team1[0].players[m].single + "</td><th>" + team1[0].players[m].double + "</td><th>" + team1[0].players[m].triple + "</td><th>" + team1[0].players[m].homerun + "</td><th>" + team1[0].players[m].strikeout + "</td><th>" + team1[0].players[m].walk + "</td></tr>"
  document.getElementById("team1-players").innerHTML += feedInPlayers;
}
for (let i = 0; i < 9; i++) {
  feedInPlayersTeam1()
}


// just do the same for team 2
let n = -1
function feedInPlayersTeam2() {
  n += 1;
  feedInPlayers = "<tr><th class='player-name'>" + team2[0].players[n].name + "</th><th>" + team2[0].players[n].hits + "</th><th>" + team2[0].players[n].single + "</td><th>" + team2[0].players[n].double + "</td><th>" + team2[0].players[n].triple + "</td><th>" + team2[0].players[n].homerun + "</td><th>" + team2[0].players[n].strikeout + "</td><th>" + team1[0].players[n].walk + "</td></tr>"
  document.getElementById("team2-players").innerHTML += feedInPlayers;
}
for (let i = 0; i < 9; i++) {
  feedInPlayersTeam2()
}

function pitchingStats() {
  team1Pitcher1 = "<tr><th class='pitcher-name'>" + team1[0].pitchers[0].name + "</th><th>" + team1[0].pitchers[0].pitches + "</th><th>" + team1[0].pitchers[0].strikeout + "</th><th>" + team1[0].pitchers[0].walk + "</th><tr>"
  document.getElementById("team1Pitchers").innerHTML += team1Pitcher1;
  team1Pitcher2 = "<tr><th class='pitcher-name'>" + team1[0].pitchers[1].name + "</th><th>" + team1[0].pitchers[1].pitches + "</th><th>" + team1[0].pitchers[1].strikeout + "</th><th>" + team1[0].pitchers[1].walk + "</th><tr>"
  document.getElementById("team1Pitchers").innerHTML += team1Pitcher2;
  // for team 2 now
  team2Pitcher1 = "<tr><th class='pitcher-name'>" + team2[0].pitchers[0].name + "</th><th>" + team2[0].pitchers[0].pitches + "</th><th>" + team2[0].pitchers[0].strikeout + "</th><th>" + team2[0].pitchers[0].walk + "</th><tr>"
  document.getElementById("team2Pitchers").innerHTML += team2Pitcher1;
  team2Pitcher2 = "<tr><th class='pitcher-name'>" + team2[0].pitchers[1].name + "</th><th>" + team2[0].pitchers[1].pitches + "</th><th>" + team2[0].pitchers[1].strikeout + "</th><th>" + team2[0].pitchers[1].walk + "</th><tr>"
  document.getElementById("team2Pitchers").innerHTML += team2Pitcher2;
}

let inningsReadable = 0;
let pitchesPerInning = 0;
let runsPerInning = 0;
let hitsPerInning = 0;

function gameRecap(){
  inningsReadable = inning / 2;
  inningsReadable += -0.5;
  totalPitches = team1[0].pitches + team2[0].pitches;
  pitchesPerInning = totalPitches / inningsReadable;
  pitchesPerInning = pitchesPerInning.toFixed(1)
  runsPerInning = team1[0].runs + team2[0].runs;
  runsPerInning = runsPerInning / inningsReadable;
  runsPerInning = runsPerInning.toFixed(1);
  hitsPerInning = team1[0].hits + team2[0].hits;
  hitsPerInning = hitsPerInning / inningsReadable;
  hitsPerInning = hitsPerInning.toFixed(1);
  document.getElementById("inningsDisplayed").innerHTML += "<h3 class='innings'>Innings: " + inningsReadable + "</h3>";
  document.getElementById("miscstats").innerHTML += "<p class='indepth'>Total pitches: " + totalPitches + "</p>";
  document.getElementById("miscstats").innerHTML += "<p class='indepth'>Avg pitches per inning: " + pitchesPerInning + "</p>";
  document.getElementById("miscstats").innerHTML += "<p class='indepth'>Avg runs per inning: " + runsPerInning + "</p>";
  document.getElementById("miscstats").innerHTML += "<p class='indepth'>Team 1 hits: " + team1[0].hits + "</p>";
  document.getElementById("miscstats").innerHTML += "<p class='indepth'>Team 2 hits: " + team2[0].hits + "</p>";
  document.getElementById("miscstats").innerHTML += "<p class='indepth'>Avg hits per inning: " + hitsPerInning + "</p>";
}

gameRecap()

pitchingStats()

console.log(team1[0])
console.log(team2[0])